pkg/services/ngalert/state/historian/annotation_test.go

```diff
@@ -40,16 +46,85 @@ func TestAnnotationHistorian_Integration(t *testing.T) {
 			require.Equal(t, frame.Fields[i].Len(), 1)
 		}
 	})
+
+	t.Run("writing state transitions as annotations succeeds", func(t *testing.T) {
+		anns := createTestAnnotationBackendSut(t)
+		rule := createTestRule()
+		states := singleFromNormal(&state.State{
+			State:  eval.Alerting,
+			Labels: data.Labels{"a": "b"},
+		})
+
+		err := <-anns.RecordStatesAsync(context.Background(), rule, states)
+
+		require.NoError(t, err)
+	})
+
+	t.Run("emits expected write metrics", func(t *testing.T) {
+		reg := prometheus.NewRegistry()
+		met := metrics.NewHistorianMetrics(reg)
+		anns := createTestAnnotationBackendSutWithMetrics(t, met)
+		errAnns := createFailingAnnotationSut(t, met)
+		rule := createTestRule()
+		states := singleFromNormal(&state.State{
+			State:  eval.Alerting,
+			Labels: data.Labels{"a": "b"},
+		})
+
+		<-anns.RecordStatesAsync(context.Background(), rule, states)
+		<-errAnns.RecordStatesAsync(context.Background(), rule, states)
+
+		exp := bytes.NewBufferString(`
+# HELP grafana_alerting_state_history_transitions_failed_total The total number of state transitions that failed to be written - they are not retried.
+# TYPE grafana_alerting_state_history_transitions_failed_total counter
+grafana_alerting_state_history_transitions_failed_total{org="1"} 1
+# HELP grafana_alerting_state_history_transitions_total The total number of state transitions processed by the state historian.
+# TYPE grafana_alerting_state_history_transitions_total counter
+grafana_alerting_state_history_transitions_total{org="1"} 2
+# HELP grafana_alerting_state_history_writes_failed_total The total number of failed writes of state history batches.
+# TYPE grafana_alerting_state_history_writes_failed_total counter
+grafana_alerting_state_history_writes_failed_total{org="1"} 1
+# HELP grafana_alerting_state_history_writes_total The total number of state history batches that were attempted to be written.
+# TYPE grafana_alerting_state_history_writes_total counter
+grafana_alerting_state_history_writes_total{org="1"} 2
+`)
+		err := testutil.GatherAndCompare(reg, exp,
+			"grafana_alerting_state_history_transitions_total",
+			"grafana_alerting_state_history_transitions_failed_total",
+			"grafana_alerting_state_history_writes_total",
+			"grafana_alerting_state_history_writes_failed_total",
+		)
+		require.NoError(t, err)
+
+		require.NoError(t, err)
+	})
 }
 
 func createTestAnnotationBackendSut(t *testing.T) *AnnotationBackend {
+	return createTestAnnotationBackendSutWithMetrics(t, metrics.NewHistorianMetrics(prometheus.NewRegistry()))
```

Comment:
nit: this abstraction feels unnecessary it's pretty common just to return the registry as a separate argument.

```golang
createTestAnnotationBackendSut(t *testing.T) (*AnnotationBackend, prometheus.Registry)
```

or make `createTestAnnotationBackendSut` accept a registry.

Reply:
I can't return the registry as a separate arg - specifically, I want to create multiple backends using the same registry.

It's super minor, I agree. I added this is so that the other (non-metrics) tests don't have to spend an extra line to construct a registry, it's basically just an ergonomics thing.

I expect us to only have 1 or 2 metrics tests, but a larger number of other tests. All the other tests don't really care about the registry, so this removes a line of visual clutter in all of them. I agree that I probably wouldn't have used an abstraction like this in production (non-test) code, but IMO tests are more about visual clarity.